components/base/sensorcontrolled/movestraight.go

```diff
@@ -54,6 +57,26 @@ func (sb *sensorBase) MoveStraight(
 		}
 	}
 
+	// if loop has been tuned but the values haven't been added to the config, error with tuned values
+	switch {
+	case (sb.configPIDVals[0].NeedsAutoTuning() && !(*sb.tunedVals)[0].NeedsAutoTuning()) &&
+		(sb.configPIDVals[1].NeedsAutoTuning() && !(*sb.tunedVals)[1].NeedsAutoTuning()):
+		err := "\"control_parameters\": ["
+		err += control.TunedPIDErr((*sb.tunedVals)[0]) + ", "
+		err += control.TunedPIDErr((*sb.tunedVals)[1]) + "]"
+		return fmt.Errorf("%v has been tuned, copy these control parameters into the config to enable movement: %v",
+			sb.Name().ShortName(), err)
+	case sb.configPIDVals[0].NeedsAutoTuning() && !(*sb.tunedVals)[0].NeedsAutoTuning():
+		err := control.TunedPIDErr((*sb.tunedVals)[0])
+		return fmt.Errorf("%v has been tuned, copy these linear control parameters into the config to enable movement: %v",
+			sb.Name().ShortName(), err)
+	case sb.configPIDVals[1].NeedsAutoTuning() && !(*sb.tunedVals)[1].NeedsAutoTuning():
+		err := control.TunedPIDErr((*sb.tunedVals)[1])
+		return fmt.Errorf("%v has been tuned, copy these angular control parameters into the config to enable movement: %v",
+			sb.Name().ShortName(), err)
+	default:
+	}
+
```

Comment:
heres how this code would look with my recommended change. 
```suggestion
if err := control.CheckTuning(sb.Name().Name, sb.configPIDVals, *sb.tunedVals); err != nil {
		return err
	}
```

Reply:
Let;s find a better way to code this for maintainability and readability.